presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestHistogram.java

```diff
@@ -80,11 +83,7 @@
     public void testSimpleHistograms()
     {
         MapType mapType = mapType(VARCHAR, BIGINT);
-        InternalAggregationFunction aggregationFunction = getMetadata().getFunctionRegistry().getAggregateFunctionImplementation(
-                new Signature(NAME,
-                        AGGREGATE,
-                        mapType.getTypeSignature(),
-                        parseTypeSignature(StandardTypes.VARCHAR)));
+        InternalAggregationFunction aggregationFunction = getInternalAggregation(mapType.getTypeSignature(), parseTypeSignature(StandardTypes.VARCHAR));
```

Comment:
let's talk in person--i thought i collapsed all the call cases, but maybe i missed some

Reply:
I renamed

the lowest level, taking two params, is "getAggregation"
then, as it was before, 

"getDefaultVarCharAggregation" (really a map varhcar -> #, but all histograms return a map, and histograms specialize only in one type,, they "data" or "key").

all I'm saying is leaving VarChar is fine by me and for those understand the type system, is enough info to infer all bindings.

Reply:
oh, you could collapse it, but then it would be off compared to the rest of the function. it goes through various key types. using the getDefaultVarCharAggregattion for one and the rest specifying the keys explicitly seems odd
i will leave this